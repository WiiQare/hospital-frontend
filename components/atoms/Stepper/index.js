import React, { useContext, useEffect } from "react";
import { FormContext } from "../../../pages/voucher/buy";

function Stepper() {
  const { activeStepIndex } = useContext(FormContext);

  useEffect(() => {
    const stepperItems = document.querySelectorAll(".stepper-item");
    stepperItems.forEach((step, i) => {
      if (i <= activeStepIndex) {
        step.classList.add("bg-primary", "text-white");

        if (i < activeStepIndex)
          step.nextSibling.classList.add("border-primary");
      } else {
        step.classList.remove("bg-primary", "text-white");
      }
    });
  }, [activeStepIndex]);

  return (
    <div className="w-full md:w-2/3 flex flex-row items-center justify-center py-4 mb-14 md:mb-16">
      <div className="flex flex-col relative">
        <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
          1
        </div>
        <span className="absolute -bottom-6 text-xs w-max md:-left-8">
          Identity <span className="hidden md:inline">For Receiver</span>{" "}
        </span>
      </div>

      <div className="flex-auto border-t-2"></div>

      <div className="flex flex-col relative">
        <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
          2
        </div>
        <span className="absolute -bottom-6 text-xs w-max md:-left-8">
          Amount <span className="hidden md:inline">to be send</span>{" "}
        </span>
      </div>
      <div className="flex-auto border-t-2"></div>
      <div className="flex flex-col relative">
        <div className="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full">
          3
        </div>
        <span className="absolute -bottom-6 text-xs w-max md:-left-8">
          Share<span className="hidden md:inline"> QR Code Voucher</span>{" "}
        </span>
      </div>
    </div>
  );
}

export default Stepper;
