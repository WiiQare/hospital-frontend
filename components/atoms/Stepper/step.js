import React, { useContext } from "react";
import { FormContext } from "../../../pages/voucher/buy";
import Identity from "./Forms/identity";
import Payment from "./Forms/payment";
import Send from "./Forms/send";

function Step() {
  const { activeStepIndex } = useContext(FormContext);
  let stepContent;

  switch (activeStepIndex) {
    case 0:
      stepContent = <Identity />;
      break;

    case 1:
      stepContent = <Payment />;
      break;
    case 2:
      stepContent = <Send />;
      break;
    default:
      break;
  }

  return stepContent;
}

export default Step;
