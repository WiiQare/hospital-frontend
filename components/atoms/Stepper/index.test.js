import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Stepper from ".";
import { FormContext } from "../../../pages/voucher/buy";

describe("Stepper", () => {
  let component;
  beforeEach(() => {
    const res = render(
      <FormContext.Provider value={{ activeStepIndex: 0 }}>
        <Stepper />
      </FormContext.Provider>
    );
    component = res.container;
  });

  it("should render the component", () => {
    expect(component).toMatchSnapshot();
  });
});
