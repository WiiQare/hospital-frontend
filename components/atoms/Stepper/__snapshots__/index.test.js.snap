// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper should render the component 1`] = `
<div>
  <div
    class="w-full md:w-2/3 flex flex-row items-center justify-center py-4 mb-14 md:mb-16"
  >
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full bg-primary text-white"
      >
        1
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Identity 
        <span
          class="hidden md:inline"
        >
          For Receiver
        </span>
         
      </span>
    </div>
    <div
      class="flex-auto border-t-2"
    />
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full"
      >
        2
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Amount 
        <span
          class="hidden md:inline"
        >
          to be send
        </span>
         
      </span>
    </div>
    <div
      class="flex-auto border-t-2"
    />
    <div
      class="flex flex-col relative"
    >
      <div
        class="stepper-item flex items-center justify-center w-12 h-12 text-center font-medium border-2 rounded-full"
      >
        3
      </div>
      <span
        class="absolute -bottom-6 text-xs w-max md:-left-8"
      >
        Share
        <span
          class="hidden md:inline"
        >
           QR Code Voucher
        </span>
         
      </span>
    </div>
  </div>
</div>
`;
